app/src/main/java/com/kickstarter/viewmodels/PrivacyViewModel.kt

```diff
@@ -0,0 +1,160 @@
+
+            this.currentUser.observable()
+                    .take(1)
+                    .compose(bindToLifecycle())
+                    .subscribe({ this.userOutput.onNext(it) })
+
+            this.userInput
+                    .concatMap<User>({ this.updateSettings(it) })
+                    .compose(bindToLifecycle())
+                    .subscribe({ this.success(it) })
+
+            this.userInput
+                    .compose(bindToLifecycle())
+                    .subscribe(this.userOutput)
+
+            this.userOutput
+                    .window(2, 1)
+                    .flatMap<List<User>>({ it.toList() })
+                    .map<User>({ ListUtils.first(it) })
+                    .compose<User>(Transformers.takeWhen<User, Throwable>(this.unableToSavePreferenceError))
+                    .compose(bindToLifecycle())
+                    .subscribe(this.userOutput)
+
+            this.optIntoFollowing
+                    .compose<Boolean>(bindToLifecycle<Boolean>())
+                    .filter({ checked -> checked })
+                    .subscribe({ _ -> this.userInput.onNext(this.userOutput.value.toBuilder().social(true).build()) })
+
+            this.optIntoFollowing
+                    .compose<Boolean>(bindToLifecycle<Boolean>())
+                    .filter({ checked -> !checked })
+                    .subscribe({ _ -> this.showConfirmFollowingOptOutPrompt.onNext(null) })
+
+            this.optOutOfFollowing
+                    .compose<Boolean>(bindToLifecycle<Boolean>())
+                    .filter({ optOut -> optOut })
+                    .subscribe({ _ -> this.userInput.onNext(this.userOutput.value.toBuilder().social(false).build()) })
+
+            this.optOutOfFollowing
+                    .compose<Boolean>(bindToLifecycle<Boolean>())
+                    .filter({ optOut -> !optOut })
+                    .subscribe({ _ -> this.hideConfirmFollowingOptOutPrompt.onNext(null) })
+        }
+
+        override fun optIntoFollowing(checked: Boolean) {
+            this.optIntoFollowing.onNext(checked)
+        }
+
+        override fun optOutOfFollowing(optOut: Boolean) {
+            this.optOutOfFollowing.onNext(optOut)
+        }
+
+        override fun optedOutOfRecommendations(checked: Boolean) {
+            this.userInput.onNext(this.userOutput.value.toBuilder().optedOutOfRecommendations(!checked).build())
+        }
+
+        override fun showPublicProfile(checked: Boolean) {
+            this.userInput.onNext(this.userOutput.value.toBuilder().showPublicProfile(!checked).build())
+        }
+
+        override fun hideConfirmFollowingOptOutPrompt(): Observable<Void> {
+            return this.hideConfirmFollowingOptOutPrompt
+        }
+
+        override fun showConfirmFollowingOptOutPrompt(): Observable<Void> {
+            return this.showConfirmFollowingOptOutPrompt
+        }
+
+        override fun user(): Observable<User> {
+            return this.userOutput
```

Comment:
Everything looks good completely up to you but you could also use type inference here 🙃
```
override fun showConfirmFollowingOptOutPrompt() = this.showConfirmFollowingOptOutPrompt

override fun user(): Observable<User> = this.userOutput
```